### DISCUSSION TITLE
Remove reodering of quest types

### DISCUSSION THREAD

**Discussion by westnordost** [streetcomplete/StreetComplete#4485] [General]
### Reordering of quests, why it exists

So, there exists for a very long time already this advanced feature that you can in the quest selection screen reorder the quest types.

This was handy back before [v26.0](https://github.com/streetcomplete/StreetComplete/releases/tag/v26.0) when each quest type was downloaded separately in the given order: Since the download of all quests took ages, the app would only download 5 or so quest types in a row before only continuing when a certain threshold of unsolved quests was undercut. For this reason, the default order of the quest types was also more or less ordered by perceived importance. So, people could select which quest types they wanted to solve first, i.e. which would be most important for them to complete.

This changed in v26. Now, all quest types are downloaded in one go, so that use case is gone. The use case that remains for quest ordering is that it determines in which order the quests are displayed. I.e. if there are several quests too close to each other, only the one further up in the quest order list is shown.
Now, however, the default quest order is not based on (perceived) importance but on how quick and easy one is able to solve this from afar. E.g. whether a bench has a backrest is quite more unimportant than opening hours, but it is solvable from further away and faster, so it is higher up in the list.

### Why I plan to remove it

I plan to add a new feature (#3481) soon that will enable loading/saving a quest preset from/to an URL (QR-Code). It is easier to implement if I do not need to worry about including the reordering-information in that URL. 

Also, generally I want to keep the code base and settings / UI clean, i.e. tend to remove things that are no longer serve their purpose.

Which is why I am considering to remove it. I already considered to remove it right for or shortly after releasing v26, but promised power users who manned the barricades on hearing it would be removed to keep it for now so they can find out if there is really a valid use case for those left that justifies their continued existence and not only something like this https://xkcd.com/1172/

### To summarize, the feature had the following use cases

1. *Control which quests are downloaded first*
  This use case is completely obsolete.

2. *Control which quests are displayed with priority.*
  IMO this is also obsolete because the quests *used to* be ordered not by how quick and easy they are solvable from afar because of reason 1. But now, they are, so there should be no more situations like that you could answer that easier quest but it is hidden below an "important" quest. For the case that a quest type is deemed by a user as too unimportant to bother, he can simply disable it

### Your input

I would like to know from you: 

Did you know about this feature? If yes, do you use it and what for? Also, do you maintain your own fork and/or use a fork, such as @Helium314's extended edition?

**Comment by FloEdelmann** [comment]
I knew about this and have used it before. But I haven't touched the default order for a long time now, so I think it's fair to remove the reordering feature given your reasons.

Regarding forks: I don't have my own fork, and I (very) rarely use @Helium314's expert edition.

**Comment by mnalis** [comment]
>  Now, however, the default quest order is not based on (perceived) importance but on how quick and easy one is able to solve this from afar. E.g. whether a bench has a backrest is quite more unimportant than opening hours, but it is solvable from further away and faster, so it is higher up in the list.

...and that is **exactly the problem** for me, and why I really **want the custom ordering**. The _current default order_ caters to the "gaming" population (_"hey, let's solve as many quests as possible to get higher score"_). While I do not exempt myself from gamification gratification, in my case it is only an afterthough/secondary objective. **My primary objective** is to **_map the stuff I consider very useful_** (which is mostly cycling infrastructure and related amenities like `drinking_water` and `opening_hours`, and blind/disabled people infrastructure), and SC is only a tool to accomplish that goal. 
While I'm fine with current SC ordering _default_, it is only as long as it is _only_ the _default_ -- i.e. something I can change if I want to (and I do want to -- I practically never use that default configuration).

So, given the choice between mapping just **10** `cycleways` and `drinking_waters`, _or_ **200** `trashcans at bus stops`, `bridge structures`, `traffic calming types`,  `memorial types`, `stile step counts` or `number of lanes`, I'd choose the **_former_** (even if I myself suggested some of the latter!) 

That is because I contribute to OSM mostly **not** because I am bored (or a gaming addict), but because I intend to **_actually use_** that map later for useful stuff like route planing and navigation.

---

Now don't get me wrong - _if_ I have time and opportunity, I'd _happily_ solve both that 10 _and_ that 200 quests. But most often it is the case that I can't, and then I have to prioritize - and **_I prioritize on perceived USEFULNESS to me and groups I associate with_**, and absolutely **NOT** based on how easy something is. 

Presets help partly with filtering what I want to solve in which situation (e.g. pedestrian, or in a car etc.)  but they are far from the silver bullet -- I need quest ordering too to be able to fulfill that.

Simple example:  I'm being driven in a motor vehicle, and as I have very little time to react and solve any particular quest, I choose most restrictive preset with fewest but most important (to me) and quickly easily solvable road quests - `cycleway`, `sidewalk`, `surface`, `track_type`, `smoothness`, `lit`.
But even there, there is no way I'd be able to solve all of them in most cases - it depends on how long particular segment of the road mapped is. If it goes for 1000 meters, I can solve all of them. If it goes for 200 meters, I can solve only 2-3 of them. 50 meters? Only one, and that scarcely. etc. And that is where quest order comes into play - it allows me always solve most important quest, and then as much as possible of lower-priority quests as time slot allows.

And there is **_no way that my priorities can be reconciled with priorities of other people_** belonging to other groups. Car drivers will prioritize `surface` and `maxspeed` and won't really care about `cycleway` which is absolute number one priority for me (who don't care about `maxspeed` at all). And some won't even care about what the quest _is about_ (gasp!!) but only how quickly and easily they can solve it.

Switching presets on each segment change is not doable either (even if it was one-click preset change, instead of current [9-click combo](https://github.com/streetcomplete/StreetComplete/issues/3827#issuecomment-1058071697)) due to rapid road segment changes.

This post is already getting way too long, but there is much more why I'd want it to stay: https://github.com/streetcomplete/StreetComplete/issues/3034#issuecomment-880056249, https://github.com/streetcomplete/StreetComplete/issues/1654#issuecomment-877127272,  https://github.com/streetcomplete/StreetComplete/issues/3034#issuecomment-879913416, https://github.com/streetcomplete/StreetComplete/issues/3034#issuecomment-880056249...

To answer the extra questions: 
I do have my fork which I use most of the time currently (in addition to upstream vanilla, and @Helium314 EE), but am in longer-term process of transferring my stuff and moving over to @Helium314 ExpertEdition fork, so I could disband my own. If the custom ordering were to be thrown out, I'd either have to convince @Helium314 to continue maintaining that custom ordering feature in EE fork (as my Android-fu is way too low for such task), or I'd have to abandon that effort of reducing number of forks, and settle on one custom sorting, falling back to my own fork (to at last force *one* custom order _I personally like the most_). Of course I'd then (due to custom ordering in my fork) lose the ability to use that fizzy new import/export functionality between forks, which would make me even more sad :cry: 

---

TL;DR please **_don't remove custom ordering_**. I'd love to have export/import quests functionality, if the custom ordering is the price to be paid, I'd rather give up on export/import (whose primary purpose for me would be to save and restore quest enablement status **and** ordering anyway!)

There must be a way to preserve **both** the `quest ordering` and the `enable status`! Pretty please?

**Reply by westnordost** [reply]
Well, that's what I meant with going on the barricades. Anyway, I read your text and found what I was looking for in your response.

And I am sorry but your use case is IMO sufficiently covered by "create a preset for only the important things + a preset for all the things". Reading between the lines, your text lets shine through that when you sort quests by your preferred importance, it bears exactly the same problems we had with the default quest order before v26 - that you do not get to answering certain easier questions because they are obscured by a more important one which takes much longer to survey while you could have conveniently answered both if the other was only shown first.

I don't know if Helium will keep this feature, but in any case you also got your own fork.

**Reply by mnalis** [reply]
While I understand why SC preferring easier but less important quests by default is good marketing for StreetComplete (i.e. _"look at the number of edits!"_), in my opinion pre-v26 **_priority-based ordering_** was actually **_more useful_** for OSM map even if **_less edits_** were produced (but they _contributed more to the actual usefulness_ of the map!). 

E.g. before people limited by time _"solved 1 important and  slow quest but skipped 3 fast and unimportant ones"_, and now the _"solve 3 fast and unimportant ones and skip 1 important quest"_ (obviously, if they're not limited, they'd just solve all quests in both cases). Good for SC edit statistics, but worse for the OSM map usefulness overall. Incidentally, It also increases the problem of users trying SC and giving up because _"it mostly asks only a lot of unimportant stuff, why would I waste time with that?"_ (if important quests are shown first however, number of "useless" quests the user perceives is much much lower, so likely not to affect them)


While I always love xkcd reference, I do not think that the example I've given above can be solved satisfactorily  without custom sorting, not that it is a worthless use case. (although only custom sorting fixes it, it might be dampened a lot with **_everyone agreeing that my idea of sorting is the best idea_** and that bicycle trumps car always, but I can forsee few disagreements there :smile:). Can you propose the way how you see it can solved without custom sorting? 
It is however unfortunate that GitHub community is just a tiny percentage of SC users, as I'm sure there are quite a few more beside me that use the custom sorting feature, and would be hurt by its removal.

Currently I have _"I'm walking in leisure"_ preset (usually in big city, or when waiting for something) , where I do indeed keep upstream SC ordering for quick quests and almost all quests enabled (regardless of their usefulness). And I have few _"important things only"_ presets (for towns, for in_car and for on_bicycle, etc.) where I have different order (and much smaller number of quests).

If I can't change your mind @westnordost  (and @Helium314 or some other experienced Android dev do not revert the change and keep maintaining the custom ordering functionality), as you noticed my only option would be to do hardcoded custom ordering in my fork. While it would solve some issues, it would create new ones:
- obviously, either way I wouldn't be able to use new export/import quests functionality in my fork
- When in moving vehicle, if I stick to SC ordering, I'd have to remove important tags (like `surface`, `sidewalk`) from my _"important preset"_, as they shadow even more important quests (like `cycleway`), and I cannot guarantee that road segments will be long enough (or driving slow enough) that I'd be able to solve both (much less several of them). Then I'd have to have other presets for each of those missed tags, and remember on which road I've solved which tags and activate matching preset only... That is obviously unusable.
- If I go with my custom ordering, I can fix that moving vehicle _"important things"_ case above, but I'd have to severly hurt / give up on using my   _"I'm walking in leisure"_ preset which is producing most of my quests (by number) -- as you noted, it would revert me to pre-v26 importance-always-comes-first ordering, which would mean I would _**solve much less quests**_ in this mode, even if continued to use it (I'd as likely just give up mapping with SC in this case). 
<details>

<summary> local folklore and meta-rant</summary>

We have a saying here _"Dabogda imao pa nemao"_ which translates to _"May the God give you (something), and then take it away"_, and it is typically used as a curse (which is not my intention here! :heart: ) - but it means to convey that is **much worse feeling to have something and then lose it, than if you'd never had it in the first place**. 

Might be useful to consider that when adding feature the users may become used to, but which is likely to be removed later - maybe it is better to refuse such feature in the first place.  :sob: 

Still manning the barricades, as hope dies last...

</details>

**Reply by Helium314** [reply]
> create a preset for only the important things + a preset for all the things

I did this, and ended up almost never using the _important_ preset because I usually want at least some quests around me, which is often not the case when using the _important_ preset. This heavily depends on the exact place... Anyway, frequently switching presets to check whether there are more quests available turned out to be too annoying, even with my quick-switch.

**Comment by mrpenner** [comment]
I have reordered quests, but I think the option of disabling quests would work just as well for me. It is difficult to do massive reordering on my phone screen anyway. Much easier to scroll through and check or uncheck quests than to drag them a long way.

**Reply by JeanFred** [reply]
I think I’m the same − I believe I reordered the stairs-related quests to get the surface one 'below' the other ones (as I find it very tricky to answer!) ; but actually I might be better off disabling that one quest that confuses me altogether.

**Reply by westnordost** [reply]
Which one exactly do you find difficult to answer?

**Reply by JeanFred** [reply]
> Which one exactly do you find difficult to answer?

The surface one. I find it already quite hard to answer for roads − I can barely tell the difference between asphalt and concrete ; and I can never decide whether something is more pebbly or compacted-y, or how whether these unjointed paving stones would qualify as sett, or cobblestone etc etc. ^_^'

And for stairs it’s almost worse − more stairs around here seem to be 'big stones' with in between asphalt (presumably, I really have a hard time saying :)) so never sure what to say.

(I have considered just disabling that quest because I am clearly hopeless at it ; but every now and then there’s some clearly brand new asphalt which I can answer)

**Comment by Helium314** [comment]
I rarely change quest orders, and my usually used preset is close to default order. But I fo have some presets with different order.
E.g. in my _car_ preset, the maxspeed, maxheight and maxweight quests at much higher priority, as it's really helpful to see those quests early (at low zoom). But I wouldn't want high priority for them in other presets.

**Comment by tapetis** [comment]
I have reordered some quests. For example, I increased the priority of the max speed quest and lowered the priority of the opening hours quest (takes too long to input). I would not be able to do this just using presets. Presets are also nested too deep in the settings menu to quickly switch between them. If the reordering functionality were to be removed, I would probably create my own fork and change the quest order in the code.

**Comment by InsertUser** [comment]
I quite like the reorder feature. I've mapped in a region where displaying housenumbers is not as common so bumping building levels above housenumber made sense for that area. I wouldn't want to disable them entirely there because they are sometimes shown. 

I think I've done a few more changes but they were a bit more for personal preference. What I'd really like is more search aliases as I've had trouble finding a quest when I didnt remember that it's listed as street rather than road etc.

**Comment by G4rdener** [comment]
I used this feature once or twice, but haven't touched it in along time. The last time i changed the order wasn't even on purpose, i just wanted to activate a quest 😅
I may have a suggestion: Maybe you could put the quests in categories for (de-)activate all e.g. public transport quests in one click.

**Comment by westnordost** [comment]
Okay, I decided the keep the feature and also include it in the QR code feature I am implementing.

The remaining use case I see is basically what Helium314 described at https://github.com/streetcomplete/StreetComplete/discussions/4485#discussioncomment-3876416 :

When you primarily want to solve a/some specific quests (i.e. you also actively look for it) but don't mind to solve others, too. In this case, it is imperative that these quest(s) are never obscured by other quests, are always on top, so you can plan your route this way.
Basically, if there were overlays for everything, e.g. if there was an overlay for max speeds, only then this use case would be completely solved.

**Reply by mnalis** [reply]
> When you primarily want to solve a/some specific quests (i.e. you also actively look for it) but don't mind to solve others, too

That was kinda what might example at https://github.com/streetcomplete/StreetComplete/discussions/4485#discussioncomment-3855314 tried to illustrate too (although it was more pronounced - it wasn't just (weaker) `I don't mind to solve others`, it was `I want to solve as much as possible of others too`):

> Simple example: I'm being driven in a motor vehicle, and as I have very little time to react and solve any particular quest, I choose most restrictive preset with fewest but most important (to me) and quickly easily solvable road quests - `cycleway`, `sidewalk`, `surface`, `track_type`, `smoothness`, `lit`.
> But even there, there is no way I'd be able to solve all of them in most cases - it depends on how long particular segment of the road mapped is. If it goes for 1000 meters, I can solve all of them. If it goes for 200 meters, I can solve only 2-3 of them. 50 meters? Only one, and that scarcely. etc. And that is where quest order comes into play - it allows me always solve most important quest, and then as much as possible of lower-priority quests as time slot allows.

I'm very happy this custom sorting will stay in! :heart: Although:

> Basically, if there were overlays for everything, e.g. if there was an overlay for max speeds, only then this use case would be completely solved

tends to push me in perpetual fear mode again :smiling_face_with_tear: (as _"implementing overlay for max speed"_ would not solve my need for custom quest sorting at all, so I'd though to mention it here again).

**Comment by biketeur** [comment]
Hi,
noch ein später Beitrag zu diesem Thema.
Ich nutze SC mit etwa 10 verschiedenen Voreinstellungen, je nach Bedarf. Die einzelnen Voreinstellungen haben zwischen 1 und ca. 25 quests.
Anfangs war mir die Quest Reihenfolge egal. Aber mit fast jedem Update kommen neue quests hinzu. Und es war mühselig, die neuen quests zu finden und zu deaktivieren. Meine Lösung war, bei den einzelnen Voreinstellungen die entsprechenden quests nach oben zu schieben. Jetzt ist es einfach, die neuen quests zu finden und zu deaktivieren.

**Comment by matkoniecz** [comment]
I have my own fork with custom order because this way it persists across reinstalls.

So both reorder feature is important to me and I would not be affected by its removal.

**Comment by westnordost** [comment]
It has been proposed to replace the quest type reordering with a simpler feature: **Favorite/pin quests**.

<img width="550" height="619" alt="image" src="https://github.com/user-attachments/assets/9132a96b-7e5c-4495-bd08-c93fc3a65f8e" />

Favorited quests will just move up to the very top in priority and also be marked as such, can be unfavorited at any time. Favorites are of course saved per edit type preset, hence you can have different favorites for each. They are also included in the QR Code to share, of course.
I think it is a better solution than the free quest reordering because the UI is simpler. No more dragging and dropping scrolling over hundreds of quests. Also, it is visible what has been moved to the top, and it can be undone easily.

https://github.com/streetcomplete/StreetComplete/pull/7121

**Reply by FloEdelmann** [reply]
Note that in the screenshot above, not favorited quests still have a drag handle, but that should be removed in #7121 completely as well.

**Reply by InsertUser** [reply]
This seems like it would be a massive step backwards to me.

**Reply by westnordost** [reply]
What's your use case?

**Reply by InsertUser** [reply]
If I'm in e.g. an area where the housenumber is less likely to be shown on the building I might bump that down below building levels as levels are more likely to be used by people giving directions or I might just be surveying in a way that means I don't see them from the front.  If all I could do was favourite I'd end up prioritising building height over business details which I think are more useful 

Or I might slightly reorder e.g. the crossing questions based on the features that I'm more likely to see. I don't remember which quest it was but I've also had days where a particular quest type seems to be dominating the display and I've bumped it down a few places to increase the variety on the screen but I still want to answer that question if there's noting more interesting nearby so I don't want it off completely.

**Reply by mnalis** [reply]
**TL;DR:** I strongly **dislike** the idea of **removing custom reordering** and replacing it with _heavily crippled_ "favorites" alternative.
 IOW, I'd much prefer keeping the status quo (or better yet, adding _"move to top/bottom/original position"_ submenu).

---

  Because:

- I use custom order to **lower the priority** of certain quest as often as I do **increase the priority** of certain other quests. Yet, the proposed "favorites" schema only allows the latter, not the former. It would also need "demote" icon to even try to to be comparable alternative
- even in [**this very discussion**](https://github.com/streetcomplete/StreetComplete/discussions/4485#discussioncomment-3855314) (and every other discussion it was mentioned), I've **already** given several use cases why ordering is important, and on much of them only having ability to "favorite" wouldn't help. I can give even more examples where demoting is needed, but basically it is everywhere where my preference might change, e.g. (just the few of them off the top of my head):
  -  in Zagreb, I prioritize housenumbers, as I find them important and almost always signed, but on islands I deprioritize them -- not because they are not important (they are, even more so!), but because they are not signed in 95% of the cases, so it becomes way too spammy for me to have to skip it all the time. So in my "island daily" preset I put them on the bottom, so I can solve everything else before it is shown.
  - in "night" preset, I put "lit" quest with highest priority, as I don't often map at night, and it is ideal opportunity so solve it correctly. But on other presets I used for daytime, I put it on bottom of the list for cities, as it is less important to me then other tags like cycleway and pedestrian infrastructure.[^2]
  - when I am on e.g. bus, I have different ordering of quest; because I am moving too fast to be able to solve them all, and I want **multiple specific** priorities that cannot be solved just with 1-level-priorities (I'd need at least "priority 1" - "priority 5" to put most important quests first, and less important later)[^3]. IOW, just because I want e.g. 10 quests to have priority over remaining 180 quests, does **not** mean all ten are _equally_ important.
  - on my "leisure walk" preset I have complex quests like `opening_hours` and diets with higher priority, as that is when I have a plenty of time to solve it. In other (more hurried) walking presets, I put them with lowest priority, as in those cases I only want to solve them when there is nothing else present (as I have limited time for solving quest; yet I prefer doing some improvements on almost every walk).
- but even if _both_ "favorite" _and_ "demote" buttons were added, it would still **only work on simplest** of cases (i.e. when there is **exactly one** in a group which should be moved), and thus is would _still_ be much inferior in its capabilities than current behaviour.[^4]
- a **lot** of hours went into customizing (including the orderings) those dozen or so presets of mine (and of other advanced users too, I'm sure). Nuking all that effort is not to be taken lightly.[^1]
- although @Amaanprobably cared enough to suggest adding an approximate conversion for old ordering, as far as I understand it, it is to be _extremely lossy_ (i.e. it could only detect if it was moved "up", but not when it was moved "down", and especially not **their exact position** in cases when multiple quests were moved). Better then nothing, but still a huge loss
- I have a bad feeling that it would likely _also_ break existing QR codes (as the custom ordering made by them could not be preserved, as it could not be represented under a new scheme).

Instead of such crippled favorites _**replacing**_ the custom ordering we have, I would instead suggest the following:
- keep the custom ordering we have, and
- to address the issue of sometimes having to drag a quest _a lot_ (which I agree _is_ un-ergonomic), I'd suggest adding functionality to long press on the quest (or its handle only?) and have a popup menu offering to:
  - move the quest to the top
  - move the quest to the bottom
  - restore the quest to its original location

That should:
- solve the ergonomic issue of "long drags",
- while at the same time keeping full functionality of custom ordering, 
- and all that without introducing any breakage 

[^1]: we have a saying there, that there is a worst curse in a world: _"dabogda im'o pa nem'o"_ (rough translation: _"may the God give you a thing you desire,  and then take it away"_) -- as it is much more devastating then never having had it in a first place
[^2]:  and on "island daily" I disable the "lit" quest altogether, as presence of streetlamp is not at all indicative of it working -- i.e. it is _almost_ as likely to be non functional as it is to be working. But in cities it is indicative as they work in 99% of the cases, so I can answer them even during the day.
[^3]: e.g. I want firstly to mark presence of cycleways, then of sidewalks, then of surface, then of lit, then of crossing quests (where I also want to deprioritize some of them from default ordering) etc. 
[^4]: if both "favorite" and "demote" were implemented but in different way (i.e. not preserving "official ordering" but each click on "Favorite" **always** putting the quest at position number one, and each click on quest on "Demote" **always** putting at the bottom of the list -- regardless of any official/default SC ordering), then _one_ could get a desired ordering by playing [towers of Hanoi](https://en.wikipedia.org/wiki/Tower_of_Hanoi) with ordering; but I would hardly call that an UX improvement compared to current custom ordering

**Reply by Amaanprobably** [reply]
@mnalis I actually quite like your suggestion of keeping the existing custom ordering and adding quick actions for:

- Move to top
- Move to bottom
- Restore original position

The only issue I see is that `long-press/hold` is already 
used for `drag-to-reorder`, so that interaction doesn't seem available.

What would you think about exposing those same actions through a ↕️ (or another suitable reordering icon) instead (where the ⭐ icon is currently placed)?

Tapping it could open a small menu with:
- Move to top
- Move to bottom
- Reset position

The existing drag-to-reorder functionality would remain unchanged, so users would still have manual control, while these actions would provide a quick way to perform large jumps through a long list.

Would that address the issue? 

 - *I also think it wouldn't break any existing functionality (unlike `favorites`), giving me enough reason to currently lean on this approach*.

**Reply by mnalis** [reply]
> The existing drag-to-reorder functionality would remain unchanged, so users would still have manual control, while these actions would provide a quick way to perform large jumps through a long list.

> Would that address the issue?

Yes, that would be great improvement IMO, making ordering easier/faster to use, but without breaking existing custom ordering functionality!

**Reply by mnalis** [reply]
> The only issue I see is that `long-press/hold` is already used for `drag-to-reorder`, so that interaction doesn't seem available.

It seems extra button is unwanted though (https://github.com/streetcomplete/StreetComplete/pull/7121#issuecomment-5732387167)... Dunno if one of those might be possible alternatives (or if they'd be accepted if they are technically doable):

- if it is possible to distinguish long press on drag-handle vs. long-press[^1] on quest name (and use former for drag, and latter for opening submenu)
- or maybe open submenu on double-tap (or short tap) instead (and have long-press remain for dragging)? 

[^1]: or short tap?

**Reply by Amaanprobably** [reply]
> if it is possible to distinguish long press on drag-handle vs. long-press[^1] on quest name (and use former for drag, and latter for opening submenu)

The current implementation uses `reorderable items`, where the entire quest row can be long-pressed and dragged. The `drag-to-reorder` icon (*the drag handle*) is mostly a visual indicator at the moment, since you can start dragging from anywhere on the item.

We could potentially switch to a dedicated `drag handle`, but on mobile that would probably need a larger touch target to remain comfortable and reliable to use, which means giving it more space in the UI.

> or maybe open submenu on double-tap (or short tap) instead (and have long-press remain for dragging)?
> or short tap?

That could be feasible. My only concern is discoverability, as I'm not sure users would naturally think to double-tap (or short-tap) a quest to find additional actions.

And given that a more *explicit* UI solution wasn't favored in the previous discussion, I'm also not sure whether a gesture-based solution would be preferred.

Regardless, if any consensus is reached, I'd be happy to implement it.

**Reply by peternewman** [reply]
Things I couldn't do with favourites but can do with the existing full reordering, sort say diet quests based on expected/perceived ease of answering in my local area (e.g.  vegetarian is most "popular", is frequently signed in large text outside the shop, vegan and GF regularly appear on menus, Kosher is less frequent near me). But conversely I don't want diet quests at the very top of the list above everything else.

Also there are groups of quests which can be re-ordered for experienced users to be more efficient, but like #6890 for example, the default order may need to suit the widest group of users, as for example you can hear sound signals from a distance away, but can't feel/see vibration unless you're actually at the crossing when people are crossing.

I guess some hybrid where you could favourite within a type might work, but is going to be much harder to understand as a user.

**Reply by peternewman** [reply]
I guess as @mnalis alluded to, the ability to "boost" a quest to add e.g. 5 points to it's ordering, as well as favourite others to put them right at the top, might just about work, but it's going to be rather counter-intuitive and probably still not resolve all the uses cases some of us have.
